lib/extensions/ar_migration.rb

```diff
@@ -0,0 +1,45 @@
+module MigrationSyncHelper
+  def self.migrations_column_present?
+    ActiveRecord::Base.connection.columns("miq_regions").detect { |c| c.name == "migrations_ran" }
+  end
+
+  def self.wait_for_remote_region_migrations(subscription, version, wait_time = 1)
+    return unless MigrationSyncHelper.migrations_column_present?
+    region = MiqRegion.find(subscription.provider_region)
+    until region.migrations_ran.include?(version)
+      subscription.disable
+      subscription.enable
+      sleep(wait_time)
+      region.reload
+    end
+  end
+
+  def self.update_local_migrations_ran(version, direction)
+    return unless MigrationSyncHelper.migrations_column_present?
+    return unless (region = MiqRegion.my_region)
+
+    new_migrations = ActiveRecord::SchemaMigration.normalized_versions
+    new_migrations << version if direction == :up
```

Comment:
I don't know of any customer that's gone back down in production, but should we handle that case anyway?

Reply:
That is strange.  Since we are overwriting the column every time, it shouldn't be an issue.